### PR TITLE
chore: update dependabot and yarn-upgrade config

### PR DESCRIPTION
**Pull request by kaizencc**
Closes https://github.com/cdklabs/cdk-ops/issues/1942

The idea is to copy the configuration we already have for the default branch and target/checkout `v1-main` on June 1. 
It is a bit ugly but -- we are not likely to do this again to any other branch, and I can't think of a better way. I have tested the changes on my own forked repository and confirmed that we do indeed target/checkout a different branch under the new configuration.

Before this PR is merged, we should double-check and make sure that yarn-update has not diverged. Targeting `v2-main` as that will be the default branch when this PR gets merged (I assume the conflicts will be resolved when `v2-main` gets forward merged).

----

### All Submissions:

* [ ] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/master/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: .github/workflows/yarn-upgrade-v1main.yml [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    name: Yarn Upgrade
+    permissions:
+      contents: read
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Check Out
+        uses: actions/checkout@v3
+        with:
+          ref: v1-main
```
Comment (kaizen3031593) [inline .github/workflows/yarn-upgrade-v1main.yml:20]: This is 1 of 2 changes between this file and `yarn-upgrade.yml`.

File: .github/workflows/yarn-upgrade-v1main.yml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    needs: upgrade
+    permissions:
+      contents: write
+      pull-requests: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check Out
+        uses: actions/checkout@v3
+        with:
+          ref: v1-main
```
Comment (kaizen3031593) [inline .github/workflows/yarn-upgrade-v1main.yml:106]: This is the other. [`peter-evans/create-pull-request`](https://github.com/peter-evans/create-pull-request) creates a pull request to the branch checked out in the workflow.

**Review by madeline-k — CHANGES_REQUESTED** [review]
@kaizen3031593, it looks like this PR accidentally has a bunch of unrelated changes in it? Integration tests, README, CfnSpec files, etc. Can you remove them?

**Comment by kaizen3031593** [comment]
@madeline-k I think the unrelated changes has to do with v1 not being in sync with v2. The forward merge got through to `v2-main` yesterday, so updated the PR and I think the changes have gone away

**Review by madeline-k — APPROVED** [review]
Approving, but please do not remove the "do-not-merge" label

**Pull request merged**